tests/base/check_suite_test.py

```diff
@@ -223,3 +223,41 @@ def test_config():
     conf_suite_mod = BaseSuite.from_config(suite_mod)
     assert_that(conf_suite_mod.name, equal_to('Model Evaluation Suite'))
     assert_that(conf_suite_mod.checks.values(), has_length(check_amount))
+
+
+def test_results_selection_by_check_type():
```

Comment:
Would also add a test in which you perform suite.run then accessing one of the results
Can also verify what happen if you try to get a check result for a check that do not exist in the suite / check run that failed

Reply:
sorry for delays

> Would also add a test in which you perform suite.run then accessing one of the results

added

> Can also verify what happen if you try to get a check result for a check that do not exist in the suite 

you will get an empty list

> check run that failed

`CheckFailure` is also considered as a result, so a resulting list will include instances of it